ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -213,6 +216,32 @@ inline uint32_t get_per_core_factor(const Tensor& input_tensor_a, const Tensor&
     return 1;
 }
 
+inline std::vector<uint32_t> get_mutlti_dim_per_core_factor(const Tensor& input_tensor_a, const Tensor& input_tensor_b, uint32_t per_core_M, uint32_t per_core_N, uint32_t in0_block_w, uint32_t interm_cb_size) {
+    uint32_t max_l1_space = get_max_l1_space(input_tensor_a);
+    tt::DataFormat in0_data_format = tt_metal::datatype_to_dataformat_converter(input_tensor_a.get_dtype());
+    tt::DataFormat in1_data_format = tt_metal::datatype_to_dataformat_converter(input_tensor_b.get_dtype());
+    uint32_t in0_single_tile_size = tt_metal::detail::TileSize(in0_data_format);  // use as estimate for output as well
+    uint32_t in1_single_tile_size = tt_metal::detail::TileSize(in1_data_format);
+    for (uint32_t per_core_factor_m = per_core_M; per_core_factor_m > 1; per_core_factor_m /= 2) {
```

Comment:
why can't factor be 1? what's the foundation of this assumption?

Reply:
changed